chia/full_node/coin_store.py

```diff
@@ -520,22 +520,28 @@ async def _set_spent(self, coin_names: List[bytes32], index: uint32):
         if len(coin_names) == 0:
             return
 
-        updates = []
-        for coin_name in coin_names:
-            updates.append((index, self.maybe_to_hex(coin_name)))
-
         async with self.db_wrapper.write_db() as conn:
-            if self.db_wrapper.db_version == 2:
-                ret: Cursor = await conn.executemany(
-                    "UPDATE OR FAIL coin_record SET spent_index=? WHERE coin_name=? AND spent_index=0", updates
-                )
-
-            else:
-                ret = await conn.executemany(
-                    "UPDATE OR FAIL coin_record SET spent=1,spent_index=? WHERE coin_name=? AND spent_index=0",
-                    updates,
-                )
-            if ret.rowcount != len(coin_names):
+            rows_updated: int = 0
+            for coin_names_chunk in chunks(coin_names, MAX_SQLITE_PARAMETERS):
+                name_params = ",".join(["?"] * len(coin_names_chunk))
+                if self.db_wrapper.db_version == 2:
+                    ret: Cursor = await conn.execute(
+                        f"UPDATE OR FAIL coin_record INDEXED BY sqlite_autoindex_coin_record_1 "
```

Comment:
specifying the index here, I assume, is to make sure it's not using the `spent_index` index, right?
The condition `spent_index=0` isn't really supposed to be required. All these coins are supposed to be unspent (iiuc). Perhaps we could remove that condition and also drop the `INDEXED BY`, would that work or make sense?

Reply:
Yes, I ran this first without that index clause and it started to create a huge amount of i/o, then I figured the wrong index usage. 
I'm sure if we remove `spent_index` condition, we don't need `INDEXED BY` any more